### PR TITLE
Update changelist and download for 11.0 release

### PR DESCRIPTION
**Pull request by derekbruening**
The 11.0.0 release is now official so we update the download page for the new version and prepare the changelist for subsequent releases.

### CODE REVIEW

**Comment by derekbruening** [comment]
That a64 failure is weird: no logs.  Presumably sthg odd on Github's end.  Will retry (have to wait for others to finish to do that).

File: api/docs/download.dox [hunk 1]
```diff
@@ -50,29 +50,29 @@ The source code is available:
 
 For the very latest changes since the last official release, you can download \ref page_weekly_builds.
 
-The [10.0.0 release](https://github.com/DynamoRIO/dynamorio/releases/tag/release_10.0.0):
+The [11.0.0 release](https://github.com/DynamoRIO/dynamorio/releases/tag/release_11.0.0):
```
Comment (edeiana) [inline api/docs/download.dox:53]: nit: Also change https://github.com/DynamoRIO/dynamorio/blob/master/api/docs/new_release.dox#L93 to 11.0.0? (Or explicitly say it's just an example?).

Related to this, I noticed that we have both `11.0` and `11.0.0` tags, and they point to the same commit. I think we only want to keep `11.0.0` (following the other tags).

**Review by edeiana — APPROVED** [review]
`code_api|api.startstop` is now failing (xref #6928).
I'm re-running it.

File: api/docs/download.dox (hunk 1 above)
Comment (edeiana) [reply api/docs/download.dox:53]: Screenshot 2024-11-05 at 10 04 57 AM

**Pull request merged**